VERSION ""


NS_ :
	NS_DESC_
	CM_
	BA_DEF_
	BA_
	VAL_
	CAT_DEF_
	CAT_
	FILTER
	BA_DEF_DEF_
	EV_DATA_
	ENVVAR_DATA_
	SGTYPE_
	SGTYPE_VAL_
	BA_DEF_SGTYPE_
	BA_SGTYPE_
	SIG_TYPE_REF_
	VAL_TABLE_
	SIG_GROUP_
	SIG_VALTYPE_
	SIGTYPE_VALTYPE_
	BO_TX_BU_
	BA_DEF_REL_
	BA_REL_
	BA_SGTYPE_REL_
	SG_MUL_VAL_

BS_:

BU_: ECU_Steering Gateway

BO_ 100 SteeringCommand: 8 Gateway
 SG_ SteeringAngle : 0|16@1+ (-2000,0.1) [-2000|2000] "degrees" ECU_Steering
 SG_ SteeringSpeed : 16|8@1+ (0,1) [0|255] "deg/s" ECU_Steering

BO_ 200 SteeringStatus: 8 ECU_Steering
 SG_ CurrentAngle : 0|16@1+ (-2000,0.1) [-2000|2000] "degrees" Gateway
 SG_ ECU_Ready : 16|1@1+ (0,1) [0|1] "" Gateway

CM_ BU_ ECU_Steering "Steering ECU that responds to steering commands";
CM_ BU_ Gateway "Gateway that publishes steering commands";
CM_ BO_ 100 "Steering command message from gateway";
CM_ BO_ 200 "Steering status message from ECU";
CM_ SG_ 100 SteeringAngle "Desired steering angle in degrees";
CM_ SG_ 100 SteeringSpeed "Speed of steering movement";
CM_ SG_ 200 CurrentAngle "Current actual steering angle";
CM_ SG_ 200 ECU_Ready "ECU ready status";
